ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-sort/result-sort.tsx

```diff
@@ -0,0 +1,137 @@
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { hot } from 'react-hot-loader'
+import * as React from 'react'
+import styled from '../../styles/styled-components'
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+import { WithBackboneProps } from '../../container/backbone-container'
+import { Button, buttonTypeEnum } from '../button'
+
+const SortItemCollectionView = require('../../../component/sort/sort.view.js')
+const Backbone = require('backbone')
+const user = require('../../../component/singletons/user-instance.js')
+
+type Props = {
+  collection: Backbone.Collection<Backbone.Model>
+} & WithBackboneProps
+
+type State = {
+  collection: Backbone.Collection<Backbone.Model>
+}
+
+const Root = styled.div`
+  display: block;
+  padding: 10px;
+  min-width: 500px;
+
+  .editor-footer {
+    padding-top: 10px;
+  }
+
+  .footer-remove {
+    display: inline-block;
+    width: 100%;
+  }
+
+  .footer-remove-has-sort {
+    display: inline-block;
+    width: 50%;
+  }
+`
+
+export default hot(module)(
+    class ResultSort extends React.Component<Props, State> {
+      constructor(props: Props) {
+        super(props)
+        var resultSort = user
+          .get('user')
+          .get('preferences')
+          .get('resultSort')
+
+        this.state = {
+          collection: new Backbone.Collection(resultSort),
+        }
+      }
+
+      handleSort = () => {
+        user
+          .get('user')
+          .get('preferences')
+          .get('resultSort')
+      }
+
+      removeSort = () => {
+        user
+          .get('user')
+          .get('preferences')
+          .set('resultSort', undefined)
+        user
+          .get('user')
+          .get('preferences')
+          .savePreferences()
+      }
+
+      saveSort = () => {
+        var sorting = this.state.collection.toJSON()
+        user
+          .get('user')
+          .get('preferences')
+          .set('resultSort', sorting.length === 0 ? undefined : sorting)
+        user
+          .get('user')
+          .get('preferences')
+          .savePreferences()
+      }
+
+      render() {
+        const hasSort =
+          this.state.collection && this.state.collection.length > 0
+        return (
+          <Root>
+            <MarionetteRegionContainer
+              view={SortItemCollectionView}
+              viewOptions={() => ({
+                collection: this.state.collection,
+              })}
+            />
+
+            {hasSort ? (
```

Comment:
❓ Try passing props into Root and change styling based on that 
https://www.styled-components.com/docs/basics#adapting-based-on-props